### DISCUSSION TITLE
Questions for tests of pool function

### DISCUSSION THREAD

**Discussion by pengguanya** [insightsengineering/rbmi#364] [Q&A]
@gowerc @nociale 
https://github.com/insightsengineering/rbmi/blob/a45df71eeff8b3a134523a5c73e47aa986953ff2/tests/testthat/test-pool.R#L15
What is the object `mice_res1` and where was it defined? What is the purpose of this test in general terms?

https://github.com/insightsengineering/rbmi/blob/a45df71eeff8b3a134523a5c73e47aa986953ff2/tests/testthat/test-pool.R#L850-L853
It seems weird to me that values in results are reassigned and then re-checked against a new pool result. Can you explain the purpose of this test?

**Comment by gowerc** [comment]
Hey @pengguanya ,

For your first question, I think there was an issue with the installation of older versions of the `mice` package so instead of including it into our suggests I just pre-computed the value and then stored the raw value in the project (rather than the code to create it). The raw values are populated in `tests/testthat/helper-mice.R`  (note that all files in this folder with the name `helper-*.R` are loaded in advance of running any of the test files and as such can be used to define objects/data that are available for all subsequent test scripts). 

Ideally I should have saved the code used to produce the object....

**Comment by gowerc** [comment]
To your second question, this test is attempting to show two separate things,

1) the first result value is used as the point estimate,  i.e. the point estimate is equal exactly to the first result
2) the following tests show that the first result is ONLY used for the point estimate, i.e. having changed the first result the remaining SE/df terms are not affected. 

Let me now if you need any more clarity on this.
